lib/uri.ml

```diff
@@ -395,7 +395,8 @@ module Path = struct
       | [] when ascension > 0 -> List.rev_append
         ("/"::(rev_interject "/" Array.(to_list (make ascension "..")))) outp
       | [] -> List.(if length outp > 0 && hd outp = "/" then tl outp else outp)
-      | "/"::s::r when ascension > 0 -> loop (ascension - 1) outp r
+      | "/"::s::r when ascension > 0 ->
+        (if s = "/" then (s::r) else r) |> loop (ascension - 1) outp
```

Comment:
We can't use `|>` due to legacy OCaml version support. If you could also put the `if` into a prior pattern like

```ocaml
| "/"::"/"::r when ascension > 0 -> loop (ascension - 1) outp ("/"::r)
| "/"::_::r when ascension > 0 -> loop (ascension - 1) outp r
```

that would be great.